### ISSUE TITLE
Incorrect arguments supplied to json.JSONEncoder.default

### ISSUE THREAD

**Issue by mgrandi** [dropbox/pygerduty#49]
This happens when I try to trigger an incident with a non JSON serializable object in the 'details' dictionary:

```
Traceback (most recent call last):
  File "/home/mgrandi/rabbitmq_monitor/app/rabbitmq_monitor/pagerduty.py", line 56, in _trigger_incident
    client=self.client_name)
  File "/home/mgrandi/.pyenv/versions/rabbitmq_monitor_venv/lib/python3.5/site-packages/pygerduty/__init__.py", line 626, in trigger_incident
    client=client, client_url=client_url, contexts=contexts)
  File "/home/mgrandi/.pyenv/versions/rabbitmq_monitor_venv/lib/python3.5/site-packages/pygerduty/__init__.py", line 588, in create_event
    data=_json_dumper(data).encode('utf-8'),
  File "/home/mgrandi/.pyenv/versions/3.5.2/lib/python3.5/json/__init__.py", line 237, in dumps
    **kw).encode(obj)
  File "/home/mgrandi/.pyenv/versions/3.5.2/lib/python3.5/json/encoder.py", line 198, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/home/mgrandi/.pyenv/versions/3.5.2/lib/python3.5/json/encoder.py", line 256, in iterencode
    return _iterencode(o, 0)
  File "/home/mgrandi/.pyenv/versions/rabbitmq_monitor_venv/lib/python3.5/site-packages/pygerduty/__init__.py", line 747, in _datetime_encoder
    return json.JSONEncoder.default(obj)
TypeError: default() missing 1 required positional argument: 'o'
```

While I realize that the error is because i'm doing something wrong, there seems to be a bug where the json.JSONEncoder.default(obj) call is not using the right arguments. The exception should be (for example)

```
>>> import uuid, json
>>> json.dumps({"lol": uuid.uuid4()})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/json/__init__.py", line 230, in dumps
    return _default_encoder.encode(obj)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/json/encoder.py", line 198, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/json/encoder.py", line 256, in iterencode
    return _iterencode(o, 0)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/json/encoder.py", line 179, in default
    raise TypeError(repr(o) + " is not JSON serializable")
TypeError: UUID('9c5c7b30-def2-4497-ba84-f49b409b072c') is not JSON serializable
>>>
```

It seems the bug is that since you are calling json.JSONEncoder.default, but you you aren't directly subclassing JSONEncoder, you need to pass 'self' in as well as 'o'

https://docs.python.org/3.5/library/json.html#json.JSONEncoder.default

**Comment by gmjosack** [comment]
Thanks for the report! I've pushed a fix up to PyPI as `pygerduty-0.35.1` that should give you the proper errors now.

**Comment by mgrandi** [comment]
I just installed the 0.35.1 version and verified that I get the correct error now. Thanks for the fast turn around =)
